test/test_time_series.jl

```diff
@@ -2582,3 +2589,17 @@ end
     IS.add_time_series!(sys, component, ts)
     ts2 = IS.get_time_series_uuid(IS.SingleTimeSeries, component, ts_name)
 end
+
+@testset "Test serialization of time series keys" begin
+    key = IS.StaticTimeSeriesKey(
+        IS.SingleTimeSeries,
+        "test",
+        Dates.now(),
+        Dates.Hour(1),
+        12,
+        Dict("scenario" => "high"),
+    )
+    data = IS.serialize(key)
```

Comment:
Next line implicitly does this on the current implementation, but maybe test that data isa dictionary?

Reply:
I will change it so that it is very obvious that the test doesn't care about what `data` is. :)

Reply:
I think the test _should_ care about what `data` is, `data` needs to be something that is trivially serializable, like a `dict` or a primitive. At a certain point `IS.serialize` falls back to just returning the struct itself and that should be tested against.

Reply:
Or the test should do a round trip through JSON writing and reading.

Reply:
All I want to do is verify that a round trip works. If someone changes the intermediate type, I don't want to have to update the test. Fair point about the default fallback functions for serialize and deserialize. That can be solved by `@test key2 !== key1`. But, that actually doesn't work.

I think our fallback functions are unsafe. They should be returning a deepcopy of the inputs. Someone would be very surprised if they called serialize on a struct that contains a Dict, changed the returned value, and then observed that the original Dict was also changed. This will add some minor cost because of copies, but I think we should do it. It's unlikely but possible that someone could do this with an ext field.

Reply:
I agree that the fallbacks are unsafe. But even if they were better, I think we need to be verifying that the result of `serialize` is something that can actually be preserved under JSON serialization. To avoid relying on the intermediate type being dict, what about something like
```julia
intermediate_serialized = IS.serialize(key)
key2 = IS.deserialize(typeof(key), JSON3.read(JSON3.write(intermediate_serialized), typeof(intermediate_serialized)))
```

? If we find a good general solution, it should be encapsulated somehow and I'll use it [here](https://github.com/NREL-Sienna/InfrastructureSystems.jl/blob/508f4cf3a71cfade8c46f32c19c3d3ca40b9380b/test/test_function_data.jl#L120) and I'm sure there are other places.